### PR TITLE
ADDED: pagination suppor fot GetChannels

### PR DESCRIPTION
**Pull request by jescarri**
Related to: https://github.com/nlopes/slack/issues/522

##### Pull Request Guidelines

These are recommendations for pull requests.  
They are strictly guidelines to help manage expectations.

##### should this be an issue instead
- [x] is it a convenience method? (no new functionality, streamlines some use case)
- [ ] exposes a previously private type, const, method, etc.
- [x] is it application specific (caching, retry logic, rate limiting, etc)
- [ ] is it performance related.

##### API changes

Since API changes have to be maintained they undergo a more detailed review and are more likely to require changes.

- no tests, if you're adding to the API include at least a single test of the happy case.
- If you can accomplish your goal without changing the API, then do so.
- dependency changes. updates are okay. adding/removing need justification.

###### examples of API changes that do not meet guidelines:
- in library cache for users. caches are use case specific.
- Convenience methods for Sending Messages, update, post, ephemeral, etc. consider opening an issue instead.

### CODE REVIEW

**Comment by jescarri** [comment]
@nlopes @james-lawrence hey guys any input here?

Thanks!

File: channels.go [hunk 1]
```diff
@@ -293,12 +293,20 @@ func (api *Client) GetChannelsContext(ctx context.Context, excludeArchived bool,
 			return nil, err
 		}
 	}
-
-	response, err := api.channelRequest(ctx, "channels.list", config.values)
-	if err != nil {
-		return nil, err
+	var channels []Channel
```
Comment (james-lawrence) [inline channels.go:296]: going to ask you to follow the apgination pattern set elsewhere in the library. see user pagination for an example.

**Review by james-lawrence — CHANGES_REQUESTED** [review]

File: channels.go (hunk 1 above)
Comment (jescarri) [reply channels.go:296]: thanks for the review, yep will try to submit something in the upcoming weeks, it's been hectic over here.

**Comment by ollieparsley** [comment]
@jescarri Channels pagination support was added in #641, so I'll close this PR.

**Pull request closed without merging**